src/engine/strat_engine/devlinks.rs

```diff
@@ -0,0 +1,118 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::collections::HashSet;
+use std::fs;
+use std::io::ErrorKind;
+use std::os::unix::fs::symlink;
+use std::path::{Path, PathBuf};
+use std::str;
+
+use super::super::errors::EngineResult;
+
+use engine::Engine;
+
+pub const DEV_PATH: &str = "/dev/stratis";
+
+/// Set up directories and symlinks under /dev/stratis based on current
+/// config. Clear out other stuff.
+// Don't just remove and recreate everything in case there are processes
+// (e.g. user shells) with the current working directory within the tree.
+pub fn setup_devlinks(engine: &Engine) -> EngineResult<()> {
+    if let Err(err) = fs::create_dir(DEV_PATH) {
+        if err.kind() != ErrorKind::AlreadyExists {
+            return Err(From::from(err));
+        }
+    }
+
+    let mut existing_dirs: HashSet<_> = fs::read_dir(DEV_PATH)?
+        .map(|dir_e| {
+                 dir_e
+                     .expect("direntry is valid")
```

Comment:
This expect() is mistaken. Here's the docs:

```This io::Result will be an Err if there's some sort of intermittent IO error during iteration.```.

Reply:
ok. let's bail if this ever happens, I guess.